src/librbd/AioObjectRequest.cc

```diff
@@ -547,8 +552,13 @@ void AbstractAioObjectWrite::send_write_op(bool write_guard)
 
   librados::AioCompletion *rados_completion =
     util::create_rados_safe_callback(this);
-  int r = m_ictx->data_ctx.aio_operate(m_oid, rados_completion, &m_write,
-                                       m_snap_seq, m_snaps);
+  int r;
+  if (m_trace && m_trace.valid())
+    r = m_ictx->data_ctx.aio_operate(m_oid, rados_completion, &m_write,
+           m_snap_seq, m_snaps, m_trace.get_info());
```

Comment:
If I do it this way I am able to capture traces with writes and reads from fio. However, unit tests run fine until 
8: [ RUN      ] TestLibRBD.TestCopy
8: using old format
8: /home/vh4x/ceph/src/test/run-rbd-unit-tests.sh: line 9:  4736 Segmentation fault      (core dumped) unittest_librbd
1/3 Test   #8: run-rbd-unit-tests.sh ............***Failed    0.40 sec
Do you have any idea of what it might be?